packages/admin/cms-admin/src/common/MasterMenu.tsx

```diff
@@ -13,68 +16,122 @@ import { RouteProps, useRouteMatch } from "react-router-dom";
 
 import { useUserPermissionCheck } from "../userPermissions/hooks/currentUser";
 
-type MasterMenuItemRoute = Omit<MenuItemRouterLinkProps, "to"> & {
+type MasterMenuItemBase = {
     requiredPermission?: string;
-    route?: RouteProps;
-    to?: string;
-    submenu?: MasterMenuItem[];
 };
 
-type MasterMenuItemAnchor = MenuItemAnchorLinkProps & {
-    requiredPermission?: string;
-};
+type MasterMenuItemRoute = MasterMenuItemBase &
+    Omit<MenuItemRouterLinkProps, "to"> & {
+        type: "route";
+        route?: RouteProps;
+        to?: string;
+    };
 
-export type MasterMenuItem = MasterMenuItemRoute | MasterMenuItemAnchor;
+type MasterMenuItemCollapsible = MasterMenuItemBase &
+    Omit<MenuCollapsibleItemProps, "children"> & {
+        type: "collapsible";
+        submenu?: Array<MasterMenuItemRoute | MasterMenuItemAnchor | MasterMenuItemCollapsible>;
+        route?: RouteProps;
+    };
+
+type MasterMenuItemAnchor = MasterMenuItemBase &
+    MenuItemAnchorLinkProps & {
+        type: "anchor";
+    };
```

Comment:
@thomasdax98 and I discussed renaming the `anchor` type to `link` as it's more common. What do you think?

Reply:
I'd actually still prefer "externalLink" because that's the only use case I can think of. For internal links you always want to use a router link. 

I think "externalLink" would be easiest to understand for everyone.

Reply:
Renamed to "externalLink" in 67bd1ca61b58fd0d839f9151a8885616aedb2efb

I can change it to "link" if you don't like it